ui/src/components/build/CreateBuildConfigDialog.js

```diff
@@ -0,0 +1,267 @@
+        </FormGroup>
+        <FormGroup>
+          <ControlLabel className="required">Username</ControlLabel>
+          <FormControl
+            type="text"
+            onChange={e => this.basicAuthState.set('username', e.target.value)}
+            value={username}
+          />
+          <HelpBlock>Username for Git authentication.</HelpBlock>
+        </FormGroup>
+        <FormGroup>
+          <ControlLabel className="required">Password or Token</ControlLabel>
+          <FormControl
+            type="password"
+            onChange={e => this.basicAuthState.set('password', e.target.value)}
+            value={password}
+          />
+          <HelpBlock>Password or token for Git authentication.</HelpBlock>
+        </FormGroup>
+      </React.Fragment>
+    );
+  };
+
+  renderSshAuthentication = () => {
+    const { name = '', privateKey = '' } = this.sshAuthState.getOrEmpty();
+    return (
+      <React.Fragment>
+        <FormGroup>
+          <ControlLabel className="required">Name</ControlLabel>
+          <FormControl type="text" onChange={e => this.sshAuthState.set('name', e.target.value)} value={name} />
+          <HelpBlock>A name for the credentials.</HelpBlock>
+        </FormGroup>
+        <FormGroup>
+          <ControlLabel className="required">SSH Private Key</ControlLabel>
+          <FormControl type="file" />
+          <HelpBlock>Upload your private SSH key file.</HelpBlock>
+          <FormControl
+            componentClass="textarea"
+            onChange={e => this.sshAuthState.set('privateKey', e.target.value)}
+            value={privateKey}
+          />
+          <HelpBlock>Private SSH key file for Git authentication.</HelpBlock>
+        </FormGroup>
+      </React.Fragment>
+    );
+  };
+
+  renderNameSection = () => {
+    const { name = '' } = this.configState.getOrEmpty();
+    return (
+      <div className="section">
+        <FormGroup className="name-field">
+          <ControlLabel className="required">Name</ControlLabel>
+          <FormControl type="text" onChange={e => this.configState.set('name', e.target.value)} value={name} />
+          <HelpBlock>A name for the build</HelpBlock>
+        </FormGroup>
+      </div>
+    );
+  };
+
+  render() {
+    const {
+      show, title, onCancel, update,
+    } = this.props;
+    return (
+      <Modal show={show} container={this}>
+        <Modal.Header>
+          <Modal.CloseButton onClick={onCancel} />
+          <Modal.Title>{title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body className="modalBody">
+          <Grid fluid>
+            <Row>
+              <Col md={10}>
+                <div className="help-block">
+                  Client Build allows you to create a mobile client binary from a git source repository.
+                </div>
+                <Form>
+                  {this.renderNameSection()}
+                  {this.renderSourceSection()}
+                  {this.renderBuildSection()}
```

Comment:
can we split those into smaller components? This file is too big and it also prevents those components to be reused.

Reply:
splitted